sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -2079,4 +2078,654 @@ case class GpuConv(num: Expression, fromBase: Expression, toBase: Expression)
+        }
+        resource_array += intPart
+        // if intString starts with 0, it must be "00000...", replace it with "0"
+        val (isZero, zeroCv) = withResource(Scalar.fromString("0")) { zero =>
+          withResource(intPart.startsWith(zero)) { isZero =>
+            (isZero.incRefCount(), ColumnVector.fromScalar(zero, cv.getRowCount.toInt))
+          }
+        }
+        val intPartZeroHandled = withResource(isZero) { isZero =>
+          withResource(zeroCv) { zeroCv =>
+            isZero.ifElse(zeroCv, intPart)
+          }
+        }
+        resource_array += intPartZeroHandled
+        val decPart = withResource(Scalar.fromString("")) { emptyString =>
+          ColumnVector.fromScalar(emptyString, cv.getRowCount.toInt)
+        }
+        resource_array += decPart
+        (intPartZeroHandled.incRefCount(), decPart.incRefCount())
+      }
+    } else {
+      withResource(cv.round(roundingScale, RoundMode.HALF_EVEN)) { rounded =>
+        withResource(rounded.castTo(DType.STRING)) { roundedStr =>
+          withResource(roundedStr.stringSplit(".", 2)) { intAndDec =>
+            (intAndDec.getColumn(0).incRefCount(), intAndDec.getColumn(1).incRefCount())
+          }
+        }
+      }
+    }
+    closeOnExcept(ArrayBuffer.empty[ColumnVector]) { resource_array =>
+      val intPartPos = closeOnExcept(decTemp) { _ =>
+        withResource(intPart) { _ =>
+          withResource(Scalar.fromString("-")) { negativeSign =>
+            withResource(Scalar.fromString("")) { emptyString =>
+              intPart.stringReplace(negativeSign, emptyString)
+            }
+          }
+        }
+      }
+      resource_array += intPartPos
+      val appendZeros = "0" * appendZeroNum
+      val appendZerosCv = closeOnExcept(decTemp) { _ =>
+        withResource(Scalar.fromString(appendZeros)) { zeroString =>
+          ColumnVector.fromScalar(zeroString, cv.getRowCount.toInt)
+        }
+      }
+      val decPart = withResource(decTemp) { _ =>
+        withResource(appendZerosCv) { _ =>
+          ColumnVector.stringConcatenate(Array(decTemp, appendZerosCv))
+        }
+      }
+      (intPartPos, decPart)
+    }
+  }
+
+  private def getParts(cv: ColumnVector, d: Int): (ColumnVector, ColumnVector) = {
+    // get int part and dec part from a column vector, int part will be set to positive
+    x.dataType match {
+      case FloatType | DoubleType => {
+        getPartsFromDouble(cv, d)
+      }
+      case DecimalType.Fixed(_, scale) => {
+        getPartsFromDecimal(cv, d, scale)
+      }
+      case IntegerType | LongType | ShortType | ByteType => {
+        closeOnExcept(ArrayBuffer.empty[ColumnVector]) { resource_array =>
+          val intPartPos = withResource(cv.castTo(DType.STRING)) { intPart =>
+            withResource(Scalar.fromString("-")) { negativeSign =>
+              withResource(Scalar.fromString("")) { emptyString =>
+                intPart.stringReplace(negativeSign, emptyString)
```

Comment:
Because sign will be handled later for all types, mainly because the float -0.0/0.0 case. I'm reconsidering to handle the sign inside `getParts`.

I updated here to `lstrip` now, looks better.